libraries/lib-track/Track.h

```diff
@@ -1167,6 +1167,9 @@ class TRACK_API TrackList final
 
    friend class Track;
 
+   ///@pre (before->IsLeader() && Find(before) != end()) || before == nullptr
```

Comment:
I would change the order of the terms of `||` and put ` ` around it so that it can really be cut and pasted unchanged as an assertion expression at the top of the function definition.